ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/ServiceToOpenAPIMapper.java

```diff
@@ -266,4 +273,17 @@ private static boolean isTreatNilableAsOptionalParameter(ServiceDeclarationNode
         }
         return true;
     }
+
+    private static void extractHateoasLinkMetadata(Project project) {
+        String packageId = project.currentPackage().packageId().id().toString();
+        project.currentPackage().moduleIds().forEach(moduleId -> {
+            Module module = project.currentPackage().module(moduleId);
+            SemanticModel semanticModel = project.currentPackage().getCompilation().getSemanticModel(moduleId);
+            HateoasMetadataVisitor hateoasMetadataVisitor = new HateoasMetadataVisitor(packageId, semanticModel);
```

Comment:
Why do we need to create a HateoasMetadataVisitor for each module? Why can't we use one visitor for the entire package?

Reply:
Accessing this semantic module for each module may be causing the test failures https://scans.gradle.com/s/znlzk5yopworq/tests/overview?outcome=FAILED. The build compiler plugin seems to be stuck in infinite access. Shall we check compatibility with this?

Reply:
@lnash94 Could you please elaborate more on this ? As per my understanding every module in a ballerina project has its own semantic model. So, it is expected to retrieve the semantic model for each module in a ballerina package.

Reply:
Fixed the issue with this commit [1] 

[1] - [4ec420266e7b4600f728d29c473b50748d77e9d9](https://github.com/ballerina-platform/openapi-tools/pull/1589/commits/4ec420266e7b4600f728d29c473b50748d77e9d9)